app/services/invoices/calculate_fees_service.rb

```diff
@@ -133,10 +133,19 @@ def create_charges_fees(subscription, boundaries)
     end
 
     def should_not_create_charge_fee?(charge, subscription)
-      (charge.pay_in_advance? || !charge.prorated?) &&
-        charge.billable_metric.recurring? &&
+      if charge.pay_in_advance?
+        condition = charge.billable_metric.recurring? &&
+          subscription.terminated? &&
+          (subscription.upgraded? || subscription.next_subscription.nil?)
```

Comment:
Layout/MultilineOperationIndentation: Align the operands of an expression in an assignment spanning multiple lines.